documentation/SupportingDocuments/Population Density API - Algorithm Initial Proposal.md

```diff
@@ -0,0 +1,33 @@
+* UEs connection records data is received pseudonymized or hashed.
+* Requested Space is received in the API customers’ request.
+### Process is as follows:
+<ol>
+<li><strong>Cleanup</strong>: Repeated records will be deleted and traffic events associated with M2M lines are filtered due to they do not contribute to the process of estimating future population density.<br>
+<strong>Output 1 </strong>&rarr; Cleaned list of UEs connection records</li><br>
+<li><strong>Attribution:</strong> A cell is associated with each user in each time interval.
+<ol><li>For mobile lines with persistent records, the last cell in which each user has been connected in each interval (considering intervals of 15 minutes) will be associated.</li>
+<li>It is common to have intervals for which a user does not show activity, for
+them it will be assumed that the user remains in the same cell in which he was last seen active.</li></ol>
+<strong>Output 2</strong> &rarr; Valid UEs per cell in each time interval</li><br><br>
+<li><strong>Counting and Aggregation:</strong> A count of users is performed per cell and interval.
+On this count, a statistical analysis is performed with the elimination of outliers.
+Records associated with cells with less than K users in each time interval (k –
+anonymity) are discarded.<br>
+<strong>Output 3</strong> &rarr;  Total number of UEs connected per cell in each time interval,
+considering privacy.</li><br>
+<li><strong>Spatial indexing (no personal data is processed):</strong> the space will be divided into units (grids), and associating the grids that overlap with its coverage area to each cell.<br>
+<strong>Output 4</strong> &rarr; Regular grid covering the required area, over the cells coverage
+areas.</li><br>
+<li><strong>Distribution and aggregation:</strong> Taking into account the coverage area of the cell, users are going to be distributed, homogeneously among the grids (currently 150m x 150m or larger) associated with the cell, and the process is repeated for all cells. As we are assuming that the distribution of users in the coverage area
+of each cell is uniform, we are introducing an error/noise in the distribution of users by time interval, which will be transferred to population density predictions that contribute to reducing the risk of user reidentification. Each grid on the map can typically be served by several cells of different technologies and frequency bands. To obtain the number of users per grid and time interval, an aggregation is performed.<br>
+<strong>Output 5</strong> &rarr; Number of UEs per grid, based on distribution in the cells covering that area</li><br>
+<li><strong>Prediction:</strong> Based on the historical information of users by grid and interval, the prediction of users by grid and interval is made for each time interval of the future. For example, the population density forecast for next Tuesday should be similar to the one observed last Tuesday. As the data becomes available, improvements
```

Comment:
"For example" The whole algorithm and every its step is an example. So. let's describe one exact solution here. Either based on the same date/time last week, or few week averaging or few weeks weighted averaging (newer data has higher weigh). But we need to describe one concrete option here.

Reply:
Described a concrete solution in step 6: "The forecast, then, will be the average of the records of the previous 4 Tuesdays at the same time frame"